src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -6,176 +6,200 @@
 
 class CompareContactPartsTest {
 
+    /*************************************
+     * Compare First And Last Name Tests *
+     *************************************/
+
     @Test
     void doNamesMatch_No_DifferentLastName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("Jonh", "Doe", "John", "Adams"));
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "John", "Adams"));
     }
 
     @Test
-    void doNamesMatch_NoDifferentFirstName() {
-        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("Jonh", "Doe", "Jane", "Doe"));
+    void doNamesMatch_No_DifferentFirstName() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch("John", "Doe", "Jane", "Doe"));
     }
 
     @Test
-    void doNamesMatch_YesCaseDoesNotMatter() {
+    void doNamesMatch_Yes_CaseDoesNotMatter() {
         assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("john", "DOE", "JOHN", "Doe"));
     }
 
     @Test
-    void doNamesMatch_YesIgnoreSpaces() {
+    void doNamesMatch_Yes_IgnoreSpaces() {
         assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" John", "Doe", "John ", " Doe  "));
     }
 
     @Test
-    void doNamesMatch_YesIgnorePunctuation() {
+    void doNamesMatch_Yes_IgnoreSpacesAndCaseInsensitive() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" JoHn", "DOE", "JOHN ", " Doe  "));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnorePunctuation() {
         assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Jane", "Doe-Adams", "Jane", "Doe Adams"));
     }
 
     @Test
-    void doNamesMatch_YesHyphenatedNames() {
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Jane", "Doe-Adams", "Jane", "Doe"));
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Jane", "Adams", "Jane", "Doe-Adams"));
+    void doNamesMatch_Maybe_HyphenatedNames() {
+        assertEquals(AnswerType.maybe, CompareContactParts.doNamesMatch("Jane", "Doe-Adams", "Jane", "Doe"));
```

Comment:
Giving a what if case: I was thinking what if the we have a case of two different last names with hyphens containing the same first part. For example... ("la-chan" and "la-shin") or ("chan-la" and "shin-la"). Two separate people, but because they both have "la" in them, the code would find a successful match.

Since there is a medium possibility of the full names not matching, shouldn't this be a maybe?

Reply:
The code would say that there is a successful match on the last name, not on the whole person.  Forget about hyphenated names... There are many people with the last name "Page" or "Bilodeau".  When we recommend matches we check all properties to try to determine if it is the same person. If the name and address are the same we can be pretty sure (but not 100% due to Jr/Sr) that it is the same person. The other thing that protects us is that if we get "match" on more than 1 contact then we can let a human pick which one to use. As I put in my original comment, I won't be surprised if we discover exceptions at some point that will force us to tweak this rule but I'd like to have some real examples before we make the code more complicated. I think it could be reasonable to say that there is some size limit for when we will match on partial last name. For example, if we find multiple names in a last name, matches on parts that are 3 or less characters result in a maybe while matches of names with over 3 characters result in a yes?